### PR TITLE
Only download sidekiq-ent and sidekiq-pro if licence ENV is set

### PR DESCRIPTION
**Pull request by omgitsbillryan**
Connects: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/7665

I went ahead and created a [`sidekiq-mock` gem](https://github.com/omgitsbillryan/sidekiq-mock) gem [based on a previous recommendation](https://github.com/department-of-veterans-affairs/vets-api/pull/1637#issuecomment-355662299), but having done so, I think it adds too much "magic" and complexity that isn't warranted.  For now, I'm comfortable with the one instance of [conditional `if Rails.env.production?` sidekiq rate-limitng](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/workers/github/create_issue_job.rb#L13). I believe it to be more explicit and easier to understand for anyone unfamiliar.

### CODE REVIEW

**Comment by omgitsbillryan** [comment]
> I'm not :100: on the affect of the dependency being removed from the Gemfile.lock on the production bundle install.

Same. Doing some discovery...

**Comment by patrickvinograd** [comment]
You can try pushing this branch and deploying to dev (which runs with `RAILS_ENV=production`) and see what it does if that's helpful. LMK if you need help navigating jenkins in that regard.

**Comment by omgitsbillryan** [comment]
Silly me - this won't work.  `Gemfile.lock` is what get's used during a `bundle install`.  As this PR stands, `sidekiq-pro` & `sidekiq-ent` simply will not get downloaded :grimacing:

**Comment by omgitsbillryan** [comment]
A conditional statement in `Gemfile` is a bad idea, because the resulting `Gemfile.lock` that developers check into git will not contain any versions of `sidekiq-ent`  & `sidekiq-pro`.

_I think the best idea at this point is to just distribute the license key and have developers update their `.bundle/config`._

A stretch idea is to host `sidekiq-ent` & `sidekiq-pro` on a gem server we own and have developers locally set a [bundle mirror config](http://bundler.io/v1.5/bundle_config.html#gem-source-mirrors-1) to mirror calls outgoing `https://enterprise.contribsys.com` to `<SOME GEM SERVER WE OWN>`.  Of course, we would have to secure said server with credentials & possibly VA VPN access... which is a pretty huge can of worms.  FWIW, I'm not in favor of this solution.

**Comment by jkassemi** [comment]
I think it's safe to distribute this license key to developers through the internal repo. 

Of course, this negatively affects the ability for folks not on the team to run the vets-api application. What if we provide a means for them to run the application given that conditional (forgive the naming):

```
  unless ENV['IS_EXTERNAL_DEV']
    source 'https://enterprise.contribsys.com/' do
      gem 'sidekiq-ent'
      gem 'sidekiq-pro'
    end
  end
```

so by default we try to install sidekiq enterprise, but still offer a limited capacity to move forward without it?

File: Gemfile [hunk 1]
```diff
@@ -120,9 +120,12 @@ group :development, :test do
 end
 
 group :production do
-  # sidekiq enterprise edition
-  source 'https://enterprise.contribsys.com/' do
-    gem 'sidekiq-ent'
-    gem 'sidekiq-pro'
+  # sidekiq enterprise requires a license key to download but is only required in production
+  # for local dev environments, regular sidekiq works fine
+  unless ENV['EXCLUDE_SIDEKIQ_ENTERPRISE']
```
Comment (omgitsbillryan) [inline Gemfile]: Settled on `EXCLUDE_SIDEKIQ_ENTERPRISE` despite the double negative'ness of it because it seems more explicit.

**Comment by omgitsbillryan** [comment]
I've opened a related PR with instructions for installing Sidekiq Enterprise:
https://github.com/department-of-veterans-affairs/vets.gov-team/pull/8303

Anyone willing to approve this PR so it may be merged?

**Comment by omgitsbillryan** [comment]
@patrickvinograd I believe Ive done that already:
https://github.com/department-of-veterans-affairs/vets.gov-team/pull/8303/files#diff-2889785b5d59fcb54738ac04ae16434fR20

**Pull request merged**